### DISCUSSION TITLE
Can not retrieve a storage bucket and get the list of the objects contained in it

### DISCUSSION THREAD

**Discussion by andreabertolini1995** [supabase/supabase#13052] [Questions]
Hello,

I am having a hard time retrieving individual buckets from my Supabase storage and a list of the objects contained in them, which is ultimately what I want to do, even though it is supposed to be pretty straight forward.
This is what I am doing:

```
supabase: Client = create_client(SUPABASE_URL, SUPABASE_KEY)
headers = {"apiKey": SUPABASE_KEY, "Authorization": f"Bearer {ACCESS_TOKEN}"}
storage_client = create_client(SUPABASE_STORAGE_URL, headers, is_async=False)

# The two commands I am having problems with
res = supabase.storage().get_bucket('misc-text')
res = supabase.storage().from_('misc-text').list()
```

For the `get_bucket` command, I get the following error:
`TypeError: SyncBucket.__init__() got an unexpected keyword argument 'file_size_limit'
`
While for the list command, it just returns an empty list. 

Any idea of how to solve these issues? Thank you in advance!

**Comment by GaryAustin1** [comment]
Your first is because a new column has been added to the bucket table and what ever type checking is going on in your client has not been updated.

The second is likely you don't have a select policy, or don't meet it for the objects table and that bucket name.

**Reply by andreabertolini1995** [reply]
Hi and thank you for your answer! 

I had a closer look to my policies and fixed them, now the second one is working.
However, could you please elaborate further on your suggestion regarding the first one? I didn't fully understand. 

Thanks again!

**Reply by GaryAustin1** [reply]
I don't know if your type error is coming from Typescript or your client, you did not say what client you are using.  I'm saying that client, or typescript typing info, has not been modified to handle some new columns added to the bucket table and so is complaining about them being returned.
